032-MLOpsFromScratch/Coach/00-prereqs.md

```diff
@@ -0,0 +1,36 @@
+# Challenge 0: Pre-requisites
+
+**[Home](../README.md)** - [Next Challenge >](./01-TimeSeriesForecasting.md)
+
+## Solution 
+
+In this challenge we'll be setting up all the tools we will need to complete our challenges, preferably done a few days before the event. Additionally, we have a [presentation](MLOps%20Lectures.pptx) that can be used to help give a high level introduction to MLOps for the students, at both a general level as well as at an Azure level.  Finally, we have a [data and code](Data_and_Code.zip) set that will need to be uploaded to the Teams channel for the students to download and reference in the upcoming challenges. 
+
+1.  Azure subscription. If you do not have one, you can sign up for a [free trial](https://azure.microsoft.com/en-us/free/).  
+
+2.  [Azure Machine Learning service workspace](https://ml.azure.com/) - It is a foundational resource in
+    the cloud that you use to experiment, train, and deploy machine learning
+    models.
+
+3.  Azure DevOps subscription. If you do not have one, you can sign up for a
+    [free account](https://azure.microsoft.com/en-us/services/devops/).
+
+    - Install [Azure DevOps Machine Learning
+      extension](https://marketplace.visualstudio.com/items?itemName=ms-air-aiagility.vss-services-azureml)
+
+    - Request Admin access to [create Service Connections](https://docs.microsoft.com/en-us/azure/devops/pipelines/library/service-endpoints?view=azure-devops&tabs=yaml) within Azure DevOps to connect with Azure ML Workspace. If the access is not granted, have admin create those service connections ahead of time. To do that, create new service connections in Project Settings for your Azure ML service and Azure Subscription using service principal. 
+    Note: Azure ML Workspace must be created first before creating Service Connections. 
+  
+4.  Python Installation, version at least \>= 3.6.5. Anaconda is more preferred
+    for Data Science tasks.
+
+    - Anaconda - <https://docs.anaconda.com/anaconda/install/windows/>
+
+    - Miniconda - <https://docs.conda.io/en/latest/miniconda.html>
+
+    - <https://www.python.org/downloads/release/python-3611/>
+
+5.  Upload [data and code](Data_and_Code.zip) to the Teams Channel for the students to access for upcoming challenges.
```

Comment:
We need to address this zip file.

It's ok that it is a zip file because you have so many files and folders, but it needs to live in the `/Student/Resources` folder.

The contents of this folder are what is dumped into the General channel Files tab for the students to download.

Reply:
+1 that this should be moved to /Student/Resources folder.

What is in this ZIP file?  Where is the original source files that make it up?  It's generally recommended to NOT store ZIP files in GitHub, but rather the actual source files.   It's then up to the Coach to create the zip file from the source files and upload it to Teams for the hack event.

As Gino noted, I won't ask you to get rid of the ZIP file now.  But please move it and consider keeping the source files in GH instead for a future update.

Reply:
@jrzyshr I was about to say the same thing about the zip but then looked at it.

The file and folder structure is very deep and nested. I think it's best to keep it as a zip and drop the zip into the Teams Files tab as they originally intended.

Reply:
Ahmed and I synced about this. We made changes as per your initial suggestions but we noticed the folder structure was too deep as Gino pointed, so now we are back to the zip structure.